src/server/node_func.c

```diff
@@ -1797,7 +1798,25 @@ set_clear_target(struct pbsnode *psourcend, resource *psourcerc, int index, int
 			ptargetrc->rs_value.at_flags |= ATR_VFLAG_TARGET;
 		else
 			ptargetrc->rs_value.at_flags &= ~ATR_VFLAG_TARGET;
+
+		if (index == ND_ATR_ResourceAvail)
+			index = ND_ATR_ResourceAssn;
+		else
+			index = ND_ATR_ResourceAvail;
+
+		ptargetrc = find_resc_entry(&pnode->nd_attr[index], psourcerc->rs_defin);
+		if (!ptargetrc) {
+			ptargetrc = add_resource_entry(&pnode->nd_attr[index], psourcerc->rs_defin);
```

Comment:
We need to ensure that an indirect resource is not been set on a node which is already a target. This is been achieved using resources_assigned flags. For that flags in both resources_available and assigned should be in sync. 
Here a corresponding resources_assigned is added and a target flag is set on it. This will not be displayed during a stat command as the attribute is not `SET` for non-consumable resources.

Reply:
In that case, just doing this when index is resources_available will not suffice? I also feel that the purpose of this function was just to set the target flag and add resources_assigned to the object. It does not fit the purpose of the function.

Reply:
There was no conventions. It was being invoked from `unset_indirect()` to clear flag for `resources_assigned`

Reply:
I strongly feel to move this logic out of this function. All these special cases make this function very special. It does not match the intent this function was written for in the first place.

Reply:
Say, I'm going to make this outside the function `set_clear_target`. I need to do the following:
1) getting the node name from the resource value.
2) Invoking find_nodebyname() to find the node structure (takes log(n) time)
3) Finding the resource entry and if not found adding the appropriate entry.
4) Setting the flag on the rescource_entry

As we need to do this in multiple places it's better to be a function. And this fucntion will look similar to `set_clear_target` but with additional complexity. I'd like to hear from you before I proceed with this method. Personally, I still feel the existing function is good enough.